### ISSUE TITLE
Implicit "this" passed through non-static inner class loses @UnderInitialization

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#409]
```
The following code fails at runtime with an NPE, but does not cause an error in 
the checker framework. I'm guessing, but I believe this is because when I call 
p.call(), the "this" which is passed through to the inner class loses it's 
@UnderInitialization.

class Callback {
  static interface Proc {
    void call();
  }

  Proc p;

  class MyProc implements Proc {
    @Override
    public void call() {
      doStuff();
    }
  }

  String foo;

  Callback() {
    p = new MyProc();
    p.call();
    foo = "hello";
  }

  void doStuff() {
    System.out.println(foo.toLowerCase());
  }

  public static void main(String[] args) {
    new Callback();
  }
}
```

Original issue reported on code.google.com by `eatnumb...@google.com` on 9 Mar 2015 at 8:41

**Comment by mernst** [comment]
I have created a smaller version of this test case and committed it as `checker/tests/nullness/Issue409.java`.

**Comment by wmdietl** [comment]
There already is a `checker/tests/nullness/init/Issue409.java`. Should the smaller test case also go into `checker/tests/nullness/init/`?

**Comment by mernst** [comment]
Thanks for pointing that out!  I have moved the smaller test case, replacing the larger one.
So. the test can now be found in `checker/tests/nullness/init/Issue409.java`.
